src/pages/academy/xpCalculation/XpCalculation.tsx

```diff
@@ -0,0 +1,249 @@
+
+  // const handleApplyFilter = () => {
+  //   if (gridApi) {
+  //     gridApi.setQuickFilter(filterText);
+  //   }
+  // };
+
+  // const handleFilterKeypress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+  //   if (event.key === 'Enter') {
+  //     handleApplyFilter();
+  //   }
+  // };
+
+  const onGridReady = (params: GridReadyEvent) => {
+    setGridApi(params.api);
+    updatePaginationState();
+  };
+
+  const Controls = () => {
+    return (
+      <div className="grading-controls">
+        <div className="GridControls ag-grid-controls">
+          <div style={{ width: '130px' }} className="left-controls"></div>
+          <div className="centre-controls">
+            <Button
+              icon={IconNames.CHEVRON_BACKWARD}
+              onClick={changePaginationView('first')}
+              minimal={true}
+              disabled={pageState.isBackDisabled}
+            />
+            <Button
+              icon={IconNames.CHEVRON_LEFT}
+              onClick={changePaginationView('prev')}
+              minimal={true}
+              disabled={pageState.isBackDisabled}
+            />
+            <Button className="pagination-details hidden-xs" disabled={true} minimal={true}>
+              <div>{`Page ${pageState.currPage} of ${pageState.maxPages}`}</div>
+              <div>{pageState.rowCountString}</div>
+            </Button>
+            <Button
+              icon={IconNames.CHEVRON_RIGHT}
+              onClick={changePaginationView('next')}
+              minimal={true}
+              disabled={pageState.isForwardDisabled}
+            />
+            <Button
+              icon={IconNames.CHEVRON_FORWARD}
+              onClick={changePaginationView('last')}
+              minimal={true}
+              disabled={pageState.isForwardDisabled}
+            />
+          </div>
+          <div className="right-controls">
+            <Button icon={IconNames.EXPORT} onClick={exportCSV}>
+              <span className="hidden-xs">Export to CSV</span>
+            </Button>
+          </div>
+        </div>
+      </div>
+    );
+  }; /* Display either a loading screen or a table with overviews. */
+  const LoadingDisplay = (
+    <NonIdealState
+      className="Grading"
+      description="Fetching XP..."
+      icon={<Spinner size={SpinnerSize.LARGE} />}
+    />
+  );
+
+  const Content = (
+    <div className="Grading">
+      <Controls />
+      <Divider />
+      <AgGridReact
+        domLayout={'autoHeight'}
+        className="ag-grid-parent ag-theme-balham"
+        columnDefs={columnDefs}
+        onGridReady={onGridReady}
+        onPaginationChanged={updatePaginationState}
+        onGridSizeChanged={resizeGrid}
+        defaultColDef={defaultColDef}
+        rowData={rowData}
+        rowHeight={30}
+        pagination={true}
+        paginationPageSize={20}
+        suppressCellSelection={true}
+        suppressMovableColumns={true}
+        suppressPaginationPanel={true}
+      />
+    </div>
+  );
```

Comment:
Ditto on abstraction and possible code duplication.

Reply:
let's KIV for the future and get this merged in first to prepare for the sem